src/components/projecteditor/editor-contract.js

```diff
@@ -145,8 +149,8 @@ export default class ContractEditor extends Component {
             alert('Error: Missing fields.');
             return;
         }
-        if(!this.contract.obj.name.match(/^([a-zA-Z0-9-_]+)$/)) {
-            alert('Illegal contract name. Only A-Za-z0-9, dash (-) and underscore (_) allowed.');
+        if(!this.contract.obj.name.match(/^([a-zA-Z0-9-_]+)$/) || this.contract.obj.name.length > 16) {
```

Comment:
@Bashlund I still believe we shouldn't make the decision for the user, restricting how long SC name can it be.